libraries/chain/database.cpp

```diff
@@ -2323,30 +2311,17 @@ namespace golos { namespace chain {
                         auto vest_created = create_vesting(author, vesting_steem);
                         auto sbd_payout = create_sbd(author, sbd_steem);
 
-                        adjust_total_payout(
-                                comment,
-                                sbd_payout.first + to_sbd(sbd_payout.second + asset(vesting_steem, STEEM_SYMBOL)),
-                                to_sbd(asset(curation_tokens, STEEM_SYMBOL)),
-                                to_sbd(asset(total_beneficiary, STEEM_SYMBOL))
-                        );
-
-                        /*if( sbd_created.symbol == SBD_SYMBOL )
-                           adjust_total_payout( comment, sbd_created + to_sbd( asset( vesting_steem, STEEM_SYMBOL ) ), to_sbd( asset( reward_tokens.to_uint64() - author_tokens, STEEM_SYMBOL ) ) );
-                        else
-                           adjust_total_payout( comment, to_sbd( asset( vesting_steem + sbd_steem, STEEM_SYMBOL ) ), to_sbd( asset( reward_tokens.to_uint64() - author_tokens, STEEM_SYMBOL ) ) );
-                           */
+                        total_payout_value = sbd_payout.first + to_sbd(sbd_payout.second + asset(vesting_steem, STEEM_SYMBOL));
+                        curator_payout_value = to_sbd(asset(curation_tokens, STEEM_SYMBOL));
+                        beneficiary_payout_value = to_sbd(asset(total_beneficiary, STEEM_SYMBOL));
 
                         // stats only.. TODO: Move to plugin...
                         total_payout = to_sbd(asset(reward_tokens.to_uint64(), STEEM_SYMBOL));
 
-                        push_virtual_operation(author_reward_operation(comment.author, to_string(comment.permlink), sbd_payout.first, sbd_payout.second, vest_created));
+                        push_virtual_operation(author_reward_operation(comment.author, to_string(comment.permlink), sbd_payout.first, sbd_payout.second, vest_created, author_tokens));
```

Comment:
You can calculate author_tokens in the social_network-plugin

Reply:
@afalaleev author_tokens is calculating in a few lines of comment cashout helper, incl. this pay_curators call:
https://github.com/GolosChain/golos/blob/golos-v0.18.4/libraries/chain/database.cpp#L2320
Also, it is used in comment cashout helper here:
https://github.com/GolosChain/golos/blob/golos-v0.18.4/libraries/chain/database.cpp#L2369
Should I move all this into the plugin?
Or I should calculate it again? 
Why do not just send author_tokens in author_reward_operation, as mentioned in your issue task?

Reply:
- author_reward_operation will increase size of account_history storage size, but not it will be not dramatically 
- author_tokens can be calculated from fields in author_reward